agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4741,6 +4745,76 @@ bool slave_thread::handle_ap_capability_query(Socket *sd, ieee1905_1::CmduMessag
     return true;
 }
 
+bool slave_thread::handle_ack_request(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+    //TODO - this is a stub handler for the purpose of controller certification testing,
+    //       will be implemented later on agent certification
+    const auto mid = cmdu_rx.getMessageId();
+    LOG(DEBUG) << "Received ACK_MESSAGE, mid=" << std::dec << int(mid);
+    return true;
+}
+
+bool slave_thread::handle_client_steering_request(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+    const auto mid = cmdu_rx.getMessageId();
+    LOG(DEBUG) << "Received CLIENT_STEERING_REQUEST_MESSAGE , mid=" << std::hex << int(mid);
+    auto steering_btm_report_tlv = cmdu_rx.addClass<wfa_map::tlvSteeringRequest>();
+    if (!steering_btm_report_tlv) {
+        LOG(ERROR) << "addClass wfa_map::tlvSteeringRequest failed";
+        return false;
+    }
+
+    // auto x = steering_btm_report_tlv->request_flags()
+    auto request_mode = steering_btm_report_tlv->request_flags().request_mode;
+    LOG(DEBUG) << "request_mode: " << request_mode;
+
+    //TODO: validate the BTM request before sending the ack.
+    // build ACK message CMDU
+    auto cmdu_tx_header = cmdu_tx.create(mid, ieee1905_1::eMessageType::ACK_MESSAGE);
+
+    if (!cmdu_tx_header) {
+        LOG(ERROR) << "cmdu creation of type ACK_MESSAGE, has failed";
+        return false;
+    }
+
+    LOG(DEBUG) << "sending ACK message back to controller";
+    if (!send_cmdu_to_controller(cmdu_tx)) {
+        LOG(ERROR) << "failed to send ACK_MESSAGE";
+        return false;
+    }
+    //TODO - Send BTM Request frame to the STA including a Neighbor Report element specifying
+    // the BSSID, Operating Class and Channel Number of the identified target BSS
+
+    //check if the Request Mode bit is set
+    if ((request_mode >> (7)) & 1) {
```

Comment:
Is this correct? I thought that request_mode was a one-bit value already. IOW, I'd expect
```suggestion
    if (request_mode) {
```

Reply:
Yes, you are right, that wad before I used the addClass, I'll change it now.